Python/bytecodes.c

```diff
@@ -4349,31 +4349,42 @@ dummy_func(
             res = PyStackRef_FromPyObjectSteal(res_o);
         }
 
-        inst(CALL_ISINSTANCE, (unused/1, unused/2, callable, self_or_null, args[oparg] -- res)) {
-            /* isinstance(o, o2) */
-            PyObject *callable_o = PyStackRef_AsPyObjectBorrow(callable);
+        op(_GUARD_CALLABLE_ISINSTANCE_NULL, (callable, null, unused[oparg] -- callable, null, unused[oparg])) {
+            DEOPT_IF(!PyStackRef_IsNull(null));
+        }
```

Comment:
Since now we know that the oparg is two, we can split out the two args and get rid of the array. Also, let's give this a better name (since it's part of the same logical family as `_GUARD_TOS_NULL` and `_GUARD_NOS_NULL`.

```suggestion
        op(_GUARD_THIRD_NULL, (null, unused, unused -- null, unused, unused)) {
            DEOPT_IF(!PyStackRef_IsNull(null));
        }
```

Reply:
Nice! I agree it makes sense to make it reusable. I also moved it just under `_GUARD_NOS_NULL` to help with discoverability